### PR TITLE
This closes #23 contact form validations bug fixed

### LINKED ISSUE #23: Fixed the bug in Contact Form
## Issue 01
When user start typing, the following warning message will pops up. The validation message should trigger when user click the **Submit** button. 
And validation message is also wrong.
Replace `Minimum 10 Characters required as (Ms Club Of Sliit)` from `Name is required` <br>
image

## Issue 02
Remove the `required` attribute from the input field, because the input field is already validated.
```
<input
   id="name"
   name="name"
   type="text"
   className={
       formErrors.name.length > 0
          ? 'form-control border-danger'
           : 'form-control'
        }
   onChange={this.onChange}
   required 
/>
````

### PR DESCRIPTION
**Pull request by Shehanx86**

### CODE REVIEW

File: src/pages/contact/contact-form-section.tsx [hunk 1]
```diff
@@ -30,9 +20,34 @@ class ContactFormSection extends React.Component<IProps, IState> {
...
   }
 
+  formValid = () => {
+    let formErrors = this.state.formErrors;
+    let name = this.state.name;
+    let email = this.state.email;
+    let message = this.state.message;
+    let valid = true;
+
+    if(name && name.length < 10){
```
Comment (rusiruavb) [inline src/pages/contact/contact-form-section.tsx:30]: No need to set a length for name. It just a required field. 
`if (name && name.length === 0)` **or**
`if (name)`
Both are correct way to check empty field.

File: src/pages/contact/contact-form-section.tsx [hunk 2]
```diff
@@ -30,9 +20,34 @@ class ContactFormSection extends React.Component<IProps, IState> {
...
+      formErrors.name = 'Name is required';
+      valid = false;
+    }
+    if(email && !emailRegEX.test(email)){
+      formErrors.email = 'Invalid email address';
+      valid = false;
+    }
+    if(message && message.length < 10){
+      formErrors.message = 'Write a message to us / Minimum 10 Characters required'
```
Comment (rusiruavb) [inline src/pages/contact/contact-form-section.tsx:39]: Change `'Write a message to us / Minimum 10 Characters required'` to `'Message must have at least 10 characters'`. And add another validation to check whether the message field is empty or not.
```
if (!message) {
   formErrors.message = 'Message is required';
   valid = false;
}
```

File: src/pages/contact/contact-form-section.tsx [hunk 3]
```diff
@@ -30,9 +20,34 @@ class ContactFormSection extends React.Component<IProps, IState> {
...
+    this.setState({ formErrors });
+    
+    return valid;
+  };
+
   handleSubmit = (event: any) => {
     event.preventDefault();
-    if (formValid(this.state.formErrors)) {
+    if (this.formValid()) {
       console.log(`----Clicked Send Button---
```
Comment (rusiruavb) [inline src/pages/contact/contact-form-section.tsx:51]: Remove the console log

**Review by rusiruavb — CHANGES_REQUESTED** [review]

File: src/pages/contact/contact-form-section.tsx (hunk 1 above)
Comment (senuravihanjayadeva) [reply src/pages/contact/contact-form-section.tsx:30]: Please consider above request

File: src/pages/contact/contact-form-section.tsx (hunk 3 above)
Comment (senuravihanjayadeva) [inline src/pages/contact/contact-form-section.tsx:51]: Remove console log

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

**Review by rusiruavb — APPROVED** [review]
LGTM

**Review by senuravihanjayadeva — APPROVED** [review]

**Pull request merged**